## OVERVIEW kasus

Perusahaan X ingin meningkatkan penjualan mereka. Dari data penjualan sebelumnya menunjukkan bahwa penjualan rata-rata yaitu $100 per transaksi. Setelah melakukan training kepada pekerja sales, data penjualan terbaru (yang diambil dari 25 sampel pekerja sales) tersimpan dalam tabel di bawah ini:

In [29]:
import pandas as pd
import numpy as np
df= np.array([100 ,150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 
              50, 100 ,100 ,140, 90 ,150, 50, 90, 120, 100, 110, 75 ,65,  ])
df

array([100, 150,  50, 100, 130, 120, 100,  85,  70, 150, 150, 120,  50,
       100, 100, 140,  90, 150,  50,  90, 120, 100, 110,  75,  65])

In [5]:
len(df)

25

In [30]:
# buat ke dataframe
data = pd.DataFrame(df, columns=['Transactions Amount'])
data

Transactions Amount
0                   100
1                   150
2                    50
3                   100
4                   130
5                   120
6                   100
7                    85
8                    70
9                   150
10                  150
11                  120
12                   50
13                  100
14                  100
15                  140
16                   90
17                  150
18                   50
19                   90
20                  120
21                  100
22                  110
23                   75
24                   65

### MEASURE OF CENTRAL TENDENCY
- MEAN
- MODE
- MEDIAN

In [70]:
import statistics
mean = statistics.mean(data['Transactions Amount'])
median = statistics.median(data['Transactions Amount'])
median = statistics.mode(data['Transactions Amount'])

print('mean:', mean)
print('median:', median)
print('modus:', mode)

mean: 102.6
median: 100
modus: [100]


In [67]:
print(mean)

None


### Measure of Variability
- Range
- variance
- Standard Deviasi

In [43]:
data.describe()

Transactions Amount
count            25.000000
mean            102.600000
std              31.823733
min              50.000000
25%              85.000000
50%             100.000000
75%             120.000000
max             150.000000

In [74]:
# atau
print('variance:',np.var(data['Transactions Amount']))
print('standard deviasi',np.std(data['Transactions Amount']))
print('range:', np.max(data['Transactions Amount'])-np.min(data['Transactions Amount']))
print('Q1:',np.quantile(data['Transactions Amount'], 0.25))
print('Q2:',np.quantile(data['Transactions Amount'], 0.5))
print('Q3:',np.quantile(data['Transactions Amount'], 0.75))

variance: 972.2399999999997
standard deviasi 31.180763300471007
range: 100
Q1: 85.0
Q2: 100.0
Q3: 120.0


### HASIL ANALISIS

#### CENTRAL of measure
- mean(102.6) > median (100) yang menunjukkan bahwa ada beberapa nilai yang lebih tinggi (outlier)
  
- Median adalah nilai tengah dari dataset. Karena median adalah 100,
  ini menunjukkan bahwa setengah dari nilai transaksi berada di bawah 100 dan setengah lainnya di atas 100

- mode (100) = menunjukan jumlah transaksi terfavorit

#### Variabilty measure
- Varians mengukur seberapa jauh nilai-nilai dalam dataset menyimpang dari rata-rata.
Jika nilai-nilai dalam dataset disini berkisar antara 0 hingga 150, maka varians 972.24 bisa dianggap tinggi...
Varians yang tinggi menunjukkan bahwa ada variasi yang signifikan dalam nilai transaksi.

- Deviasi standar adalah akar kuadrat dari varians, 
deviasi standar yang cukup besar menunjukkan bahwa ada variasi yang signifikan dalam nilai transaksi.



## DEFINE H0 dan H1 and Alpha
- H0 = sebelum training , Transasctions  Amount = 100
- H1 = setelah training, Transactions Amount > 100 (lebih besar dari 100)
- Alpha 5% (a) = 0.05
- ini berarti one sample test

### perform T test one sample test
- T statistik
- P value

In [82]:
from scipy import stats

stats.ttest_1samp(a=data['Transactions Amount'], popmean=100) #popmean = penjualan sebelum training

TtestResult(statistic=0.4085001556802841, pvalue=0.6865284813438117, df=24)

#### membandingkan 
- p value dengan alpha
- t statistik dengan critical region

###### P value and alpha
-  P value = 0.68, Alpha = 0.05
- p value > alpha, maka hasilnya gagal menolak H0.
- (hasil test kita gagal dan training sales ga guna)

###### t statistik and critical region
- cari dulu critical region nya

In [87]:
# H1 : larger (kearah kanan)

critical_region = stats.t.ppf(1-0.05 , 24) 
# 24 itu degree of freedom atau n-1 (25-1)
critical_region

1.7108820799094275

- critical region = t > 1.71 (t harus diatas 1.71)
- sedangkan t statistik hanya 0.40
- t statistik berada di luar critical region (hasil test gagal)
- training karyawan sales ga efektif